src/stores/ProfileController.ts

```diff
@@ -21,7 +33,20 @@ export const updateProfile = async ({ id, name, adult, avatar_url, pin }: Profil
 
 export const enterProfile = async ({ id, pin }: EnterProfilePayload) => {
   return await useService(async ({ profileService, sandbox }) => {
-    return await profileService?.enterProfile({ id, pin }, sandbox ?? true);
+    const response = await profileService?.enterProfile({ id, pin }, sandbox ?? true);
+    const profile = response?.responseData;
+    if (profile?.credentials?.access_token) {
```

Comment:
This can be handled on service level, and that way you will not need the condition as well